support-services/src/main/scala/com/gu/support/pricing/PriceSummaryService.scala

```diff
@@ -61,16 +61,18 @@ class PriceSummaryService(promotionService: PromotionService, catalogService: Ca
     )
   }
 
-  private def getPromotionSummary(promotion: PromotionWithCode, price: Price, billingPeriod: BillingPeriod) = {
+  private def getPromotionSummary(promotionWithCode: PromotionWithCode, price: Price, billingPeriod: BillingPeriod) = {
+    import promotionWithCode._
     PromotionSummary(
-      promotion.promotion.name,
-      promotion.promotion.description,
-      promotion.promoCode,
-      promotion.promotion.discount.map(getDiscountedPrice(price, _, billingPeriod).value),
-      promotion.promotion.discount.flatMap(_.durationMonths).map(getNumberOfDiscountedPeriods(_, billingPeriod)),
-      promotion.promotion.discount,
-      promotion.promotion.freeTrial,
-      promotion.promotion.incentive
+      name = promotion.name,
+      description = promotion.description,
+      promoCode = promoCode,
+      discountedPrice = promotion.discount.map(getDiscountedPrice(price, _, billingPeriod).value),
+      numberOfDiscountedPeriods = promotion.discount.flatMap(_.durationMonths).map(getNumberOfDiscountedPeriods(_, billingPeriod)),
```

Comment:
is this still valid for the new promotion type? do we need to update `getNumberOfDiscountedPeriods`?

Reply:
No this field only applies to discounts as does `discountedPrice`. The reason these are separated out is because they are not part of the benefit itself, they need to be worked out in conjunction with the price.